crates/node/core/src/args/static_files.rs

```diff
@@ -34,4 +44,13 @@ impl StaticFilesArgs {
             },
         }
     }
+
+    /// Converts the static files arguments into [`StorageSettings`].
+    pub fn to_settings(&self) -> StorageSettings {
+        if self.receipts {
+            StorageSettings::new().with_receipts_in_static_files()
+        } else {
+            StorageSettings::legacy()
+        }
```

Comment:
nit: think it's nicer to be able to do
```suggestion
        StorageSettings::legacy().with_receipts_in_static_files(self.receipts)
```